web/components/templates/requestsV2/costPill.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState } from "react";
+import ThemedModal from "../../shared/themed/themedModal";
+import Link from "next/link";
+import { clsx } from "../../shared/clsx";
+import { ArrowTopRightOnSquareIcon } from "@heroicons/react/24/outline";
+
+const CostPill = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleOpenModal = (
+    event: React.MouseEvent<HTMLSpanElement, MouseEvent>
+  ) => {
+    event.stopPropagation();
+    setIsOpen(true);
+  };
+
+  return (
+    <>
+      <span
+        onClick={handleOpenModal}
+        className={clsx(
+          "bg-gray-50 text-gray-700 ring-gray-200 dark:bg-gray-900 dark:text-gray-300 dark:ring-gray-800",
+          "w-max flex items-center rounded-lg px-2 py-1 -my-1 text-xs font-medium ring-1 ring-inset cursor-pointer"
+        )}
+      >
+        Cost Unsupported{" "}
+        <ArrowTopRightOnSquareIcon className="ml-1 h-4 w-4 text-orange-500 inline-block" />
+      </span>
+
+      {isOpen && (
```

Comment:
no need to add the `isOpen` check here. the ThemedModal will handle this for you. You'll notice a smoother animation once you remove this.